src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -22,16 +25,47 @@ public partial class PackageReadmeControl : UserControl
 #pragma warning disable CS0618 // Type or member is obsolete
         private IMarkdownPreview _markdownPreview;
 #pragma warning restore CS0618 // Type or member is obsolete
-
+        private bool _isInitializing = false;
         public PackageReadmeControl()
         {
             InitializeComponent();
-            _markdownPreview = MarkdownPreviewSingleton.GetInstance();
-            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
         }
 
         public ReadmePreviewViewModel ReadmeViewModel { get => (ReadmePreviewViewModel)DataContext; }
 
+        private async Task InitializeAsync()
+        {
+            if (_markdownPreview is null && !_isInitializing)
```

Comment:
This isn't thread safe which is a requirement for async methods. Wrap this in an AsyncLazy<T> or a AsyncLazyInitializer.

Reply:
I changed it so it's no longer async, the component is loaded from the PackageManagerControl and passed into the init.